src/main/java/com/google/alpollo/model/SongEntity.java

```diff
@@ -26,4 +35,21 @@ public int compare(SongEntity a, SongEntity b) {
       return Double.compare(b.salience, a.salience);
     }
   };
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
When you implement `equals`, you should also implement `hashCode`

https://www.baeldung.com/java-equals-hashcode-contracts

Reply:
Awesome, lots of new things!

Reply:
Totally forgot to mention this thing when told about `equals()`